libevmasm/Assembly.cpp

```diff
@@ -580,14 +580,14 @@ LinkerObject const& Assembly::assemble() const
 	multimap<size_t, size_t> subRef;
 	vector<unsigned> sizeRef; ///< Pointers to code locations where the size of the program is inserted
 	unsigned bytesPerTag = util::bytesRequired(bytesRequiredForCode);
-	uint8_t tagPush = (uint8_t)Instruction::PUSH1 - 1 + bytesPerTag;
+	uint8_t tagPush = (uint8_t)pushInstruction(bytesPerTag);
```

Comment:
This helper also introduces an assertion that an invalid PUSH instruction will not be generated.